### PR TITLE
Support permit extension point

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Taking over https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/pull/93.

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/issues/72

#### Special notes for your reviewer:

@kazukousen I put you in commit co-author, let me know in case you don't want to (I believe you won't 🙂).

#### Does this PR introduce a user-facing change?

```release-note
Support permit extension point
```

#### What are the benchmark results of this change?

```benchstat

```

### CODE REVIEW

File: scheduler/plugin/plugin_test.go [hunk 1]
```diff
@@ -1086,6 +1083,97 @@ func TestReserve(t *testing.T) {
...
+		},
+		{
+			name:                  "Error",
+			pod:                   test.PodSmall,
+			nodeName:              "bad",
+			expectedStatusCode:    framework.Error,
+			expectedStatusMessage: "name is bad",
+		},
+		{
+			name:                  "Error",
```
Comment (Gekko0114) [inline scheduler/plugin/plugin_test.go:1112]:
```suggestion
			name:                  "Wait",
```

**Review by Gekko0114 — COMMENTED** [review]
Overall looks good :)

BTW, how about adding an explanation somewhere that you can't pass too big int64 values as we discussed previously?
https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/pull/93/files#r1468783879

**Comment by sanposhiho** [comment]
First, changed the interface to -

```go
Permit(state CycleState, p proto.Pod, nodeName string) (status *Status, timeoutMilliSeconds uint32)
```

And, updated RATIONAL.md for the reason.

Also, actually we don't have to use hostfunction to return timeout, and so eliminated things around it.

File: scheduler/plugin/guest.go [hunk 2]
```diff
@@ -238,6 +242,21 @@ func (g *guest) unreserve(ctx context.Context) {
...
+func (g *guest) permit(ctx context.Context) (*framework.Status, time.Duration) {
+	defer g.out.Reset()
+	callStack := g.callStack
+
+	if err := g.permitFn.CallWithStack(ctx, callStack); err != nil {
+		return framework.AsStatus(decorateError(g.out, guestExportPermit, err)), 0
+	}
+
+	statusCode := int32(callStack[0] >> 32)
+	timeoutMilliSeconds := int32(callStack[0])
```
Comment (Gekko0114) [inline scheduler/plugin/guest.go:255]: How about using uint32 here, just to prevent a too big positive number from changing to a negative number?

**Comment by Gekko0114** [comment]
Thanks! Looks good and left a very minor comment.

**Comment by sanposhiho** [comment]
Updated.

**Review by Gekko0114 — APPROVED** [review]
Thanks! Left lgtm label for other reviewers

**Pull request merged**